### PR TITLE
Always use https when creating player iframe

### PR DESCRIPTION
**Pull request by myarcane**
### Description
- To avoid mix content issue always use `https:` protocol when creating player iframe

### CODE REVIEW

File: src/core/player.js [hunk 1]
```diff
@@ -151,7 +151,29 @@ DM.provide('Player',
...
+        }
+        catch(e) {
+          // current window.location fallback
+          if (window.location) {
+              location = window.location.href;
+          }
+        }
+
+        var matchingProtocol = /^https?:/.exec(location);
+        DM.Player._PROTOCOL = matchingProtocol ? matchingProtocol[0] : 'https:';
```
Comment (jakub-g) [inline src/core/player.js]: Wouldn't it be simpler to always use `https`, or there are any counter-indications?
@dharFr WDYT?

(one exception might be `file:` protocol but I don't think it's worth supporting it)

Comment (Cycymomo) [reply src/core/player.js]: at least to test locally, there is no serve yet.
You have to manually open this file in browser `file:///Dailymotion/dailymotion-sdk-js/tests/player.html`

Comment (myarcane) [reply src/core/player.js]: I guess we could maybe simplify. Worst case scenario is setting an https iframe within a parent `http` context.
Mixed content should be avoided in this case. Let's simplify ?

Comment (dharFr) [reply src/core/player.js]: Agree with @Cycymomo, the main use-case today is probably testing within a local environment (I used to use a simple local http server to test this during dev). We probably don't want to setup a certificate for a quick fix in the SDK, etc.

**Pull request merged**